### ISSUE TITLE
Some weird interactions in the Open & Share module

### ISSUE THREAD

**Issue by PabloOQ** [TrianguloY/UrlChecker#37]
When I tap on a link in the twitter app and URLChecker opens, with custom tabs 'off', if I then open the link in Firefox Focus or the DuckDuckGo browser with these apps closed, the browser opens 'inside' the twitter app, this doesn't happens with other browsers. Because of this behaviour, if I then go to another app the browser inside the twitter app closes, and I need to reopen the link. I haven't seen this happening in other apps either, only on twitter, which makes me think this is something that can be solved inside URLChecker. It would be great to have an option to avoid this, maybe a button, or a preference in the Open & Share module. Maybe even have something added to the custom tabs icon to show this, and change the behaviour by holding the button.
Some screenshots of the button concept:
- No custom tab, open externally
![image](https://user-images.githubusercontent.com/55493443/182905231-e0c055aa-1eea-4e70-90ce-300adc36420c.png)
- No custom tab, open internally
![image](https://user-images.githubusercontent.com/55493443/182905286-fd891ebd-38cc-4320-adcc-c5aeb5283284.png)
- Custom tab, open externally. 
![image](https://user-images.githubusercontent.com/55493443/182905161-f849caed-09ab-4016-aa0d-705abe4d056f.png)
- Custom tab, open internally.
![image](https://user-images.githubusercontent.com/55493443/182905132-04378912-3556-4bdc-8dc6-9bd3e8fc985b.png)

Some screenshots of the behaviour:
- URL Checker
![image](https://user-images.githubusercontent.com/55493443/182902318-830d4ef5-8039-42b2-8aaa-7557677b6388.png)
- While going to change to another tab (before the animation concludes), the "inside browser" is still open
![image](https://user-images.githubusercontent.com/55493443/182902382-7d5d33f3-c70b-413f-a2a5-c9e3302d58f5.png)
- When going to change to another tab (animation concluded), the "inside browser" is closed
![image](https://user-images.githubusercontent.com/55493443/182902449-93c182c4-9ec5-46f6-8859-704e0eb10620.png)

Also when I tap a link in the youtube app the custom tabs button is 'on' by default, but on other apps is usually 'off', the twitter app for example.  Same thing as the previously mentioned, a preference to choose the default state, 'on' or 'off' would be great, currently if feels incosistent accross apps.

**Comment by TrianguloY** [comment]
2 issues here:
- The custom tabs initial on/off is something I wanted to make. Currently the initial state is just what the app that sent the link said (it's a custom intent extra). Having a setting to choose the initial state should be relatively easy to do, in case you want to. I'll write a specific issue for it.
- the 'inside/outside' issue is a bit hard to understand. I don't have the twitter app, I'll try to test with it. Can you give some more details about what 'inside' and 'outside' mean? Maybe a step-by step guide, or a video, or something to let me understand what the issue is.

**Comment by PabloOQ** [comment]
I would love to try to add the preference for the custom tabs initial state. I won't have access to my PC for the following week, but if it's not done after that (and I don't forget) I can give it a shot.

Yeah, I didn't explain the 'inside/outside' very well, mainly because I don't really know why this happens and how is called, anyways, my bad. Here are 2 videos. As shown, the behaviour is different depending on the app.

https://user-images.githubusercontent.com/55493443/182946839-b1458874-c824-4f32-819d-a67806062705.mp4


https://user-images.githubusercontent.com/55493443/182946883-3711305d-235b-4a97-bea8-35ad2c6e6e31.mp4

**Comment by TrianguloY** [comment]
Ah, I think what that issue is. There is an android flag (FLAG_NEW_TASK or something like that) that allows apps to be opened as independent versions or as part of the parent app. Maybe that's the difference on the behavior. With the debug module enabled you should see the flags integer of the intent, most probably it will be different (you can also use the android documentation to decode it and see exactly which flags are).

A new button to enable/disable that flag could be a good idea, with the same "initial auto/on/off" configuration. The feature is different from the custom tabs, so I think it's better to keep them independent.
At the same time, another button on the open module could make it overcrowded. Maybe a new module for these "special" opening modes? Or a separated share module?

**Comment by PabloOQ** [comment]
> There is an android flag (FLAG_NEW_TASK or something like that) that allows apps to be opened as independent versions or as part of the parent app. With the debug module enabled you should see the flags integer of the intent, most probably it will be different.

I think that´s it, great! I tried looking into what this behaviour actually was for days, but because I didn't know what was actually happening I couldn't find anything.

> Maybe a new module for these "special" opening modes? Or a separated share module?

If the module were to be separated, I think the "open with" dropdown, share and copy to clipboard should be in the same, always on, module. Then, the "how" the link is going to be opened (custom tab and new task) could be in a different module, which could be deactivated, and even if deactivated it would use the preferences from the "how" module.

**Comment by TrianguloY** [comment]
Hmm, I don't like that a deactivated module can change the behavior of others. I prefer to have different modules for different features, with options to enable or disable extra things. I'll think about it.

**Comment by Efreak** [comment]
The disabled custom tabs icon is currently difficult to tell the difference. The line through it needs to be bigger or a different color.

**Comment by TrianguloY** [comment]
> The disabled custom tabs icon is currently difficult to tell the difference. The line through it needs to be bigger or a different color.

It isn't a different color to allow the system to tint it with your theme color. I think it can be made darker to differentiate though, or make it so that the diagonal line is 'separated' from the rest.

**Comment by TrianguloY** [comment]
That putData was something I added to keep extra information, you can use it to make modules share information between them without any issue, just remember that modules can be disabled so one can't depend on others. In any case if you have any suggestion for improvement feel free to test and report it!

**Comment by PabloOQ** [comment]
Closing this as completed. Also there is a dedicated issue for future improvements to the module made as a result of this issue at #202

**Comment by TrianguloY** [comment]
I guess this was due to the "don't keep activity" in the manifest, wasn't it?
My issue with the Github app disappearing from recents is now gone!

**Comment by PabloOQ** [comment]
Weird I thought it was because of the flags of those apps, that's one of the reasons I made the flags module, I think that was my conclusion when I made testing at the time, not the extra flag that URLCheck added, but you are right, it's because of the manifest. Maybe my testing was flawed? I also don't have the same phone as then, and twitter (the main culprit) has changed a lot since then.

Whatever the case, the manifest change fixes it and if the issue reappears it could be fixed with the use of flags.
